### DISCUSSION TITLE
Why not allow .at and [] to work with non json arrays?

### DISCUSSION THREAD

**Discussion by solaris783** [nlohmann/json#3059] [Q&A] [answered]
```
json j = {{"Name", "Option2"},{"Type","Float"}};
auto jstr = j[1];
```
yields error 304
In the docs it lists that using an index for a non-array doesn't make sense.  I think it does.  The example above has 2 elements.  j.size() returns 2.  So there's an index 0 and an index 1.  I want to get index 1, without having to iterate through index 0.  Say I have a json with 50 things, would I have to iterate through all 50 to get to the last element?  Why not just j[49]?
If there is a way to get an element at a particular index, please let me know.  Thanks!

**Comment by nlohmann** [comment, marked as answer]
JSON object keys are unordered (see also https://json.nlohmann.me/features/object_order/). This library uses `std::map` by default which orders them alphabetically. But just as there is no  `std::map::operator[](size_t)`, this library also does not provide such a function for JSON objects. If you want to iterate through the object keys, use normal iterators (just like in a map) or `items()` (https://json.nlohmann.me/api/basic_json/items/).
